docs/rfcs/749-unify-path-behavior.md

```diff
@@ -0,0 +1,89 @@
+  - The default value is `/`, that means the working directory is the root path for Unix, and the current volume of the process for Windows.
+  - Services SHOULD set `WorkDir` to the path name after the evaluation of any symbolic links internal.
+- For object storage, `WorkDir` is the simulated directory or prefix of the object key.
+  - `WorkDir` SHOULD be Unix style with the prefix `/`.
+  - The default value is `/`.
+  - Form service side, the prefix `/` needs to be removed for internal processing, but `/` needs to be added to the fields returned to the user.
+  
+`path` is the file path for file system, or an object key for object storage. Also, it could be a prefix filter for `List` operation.
+
+- For file system services on Windows, `path` MUST be the relative path for `WorkDir`. For file system on Unix, `path` could be an absolute path or a relative path.
+- For object storage service, `path` MUST be Unix style.
+- For the unique key `Object.ID` in storage, it SHOULD be an absolute path compatible with the target platform, unless there's a returned unique identifier like in dropbox.
+- For the path `Object.Path` in storage, it SHOULD be Unix style.
+- Users SHOULD follow the file and object naming of different services.
+
+### Path Separator
+
+Input path for users that contains system-related path separator SHOULD be allowed. When the drive letter is included for Windows, it should be something like `c:\\a\\b`.
+
+From go-storage side:
+
+- Replace each separator character in path with a slash (`/`) character could be generated for the input path.
+
+From service side:
+
+- Services SHOULD replace `/` in path to the system-related path separator at the beginning of the operation.
+
+## Rationale
+
+### filepath
+
+Package `filepath` implements utility routines for manipulating filename paths in a way compatible with the target operating system-defined file paths. `FromSlash` returns the result of replacing each slash (`/`) character in path with a separator character.
+
+### Object key
+
+- For [object key in s3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html#object-key-guidelines)
+  - Forward slash (`/`) is the safe character, and backslash (`\`) need to be avoided. There is no hierarchy of subfolders. However, users can infer logical hierarchy using key name prefixes and delimiters (`/`). 
+  - Objects with a prefix of `./` or `../`, or with key names ending with period(s) (`.`) are allowed but should be aware of the prefix limitations.
+- For [object key in oss](https://www.alibabacloud.com/help/doc-detail/87728.htm)
+  - The name cannot start with a forward slash (`/`) or a backslash (`\`).
+
+## Compatibility
+
+- `work_dir` must start with `/` is only for Unix.
+- The format for `name` and `work_dir` in connection string need to be reconsidered as we can't separate them by `/` anymore.
+
+## Implementation
+
+N/A

```

Comment:
Please give more detail of the implementation?

- Will we change the definitions?
- Will we add new integration tests?